velox/dwio/common/tests/TypeTests.cpp

```diff
@@ -29,6 +29,21 @@ using namespace facebook::velox::dwio::common::typeutils;
 using facebook::velox::type::fbhive::HiveTypeParser;
 using facebook::velox::type::fbhive::HiveTypeSerializer;
 
+void assertEqualTypeWithId(
+    std::shared_ptr<const TypeWithId>& actual,
+    std::shared_ptr<const TypeWithId>& expected) {
+  EXPECT_EQ(actual->size(), expected->size());
+  for (auto idx = 0; idx < actual->size(); idx++) {
+    auto actualTypeChild = actual->childAt(idx);
+    auto expectedTypeChild = expected->childAt(idx);
+    EXPECT_NE(actualTypeChild, expectedTypeChild);
```

Comment:
I still think this line should be removed, because as the function name suggested, it's to verify if two typeWithId trees have the same content. Other callers may pass on one single tree both as actual and expected, and in that case this function SHALL pass. This is ok because we will have another assertValidTypeWithId() to verify if a tree is still valid. Whether or not the objects are the same one is not a focus of the test.

Reply:
We want to verify that content is same but the objects are different which means they are copies rather than referencing members from each other , that is why I added this check.